packages/@aws-cdk/aws-redshift/lib/private/database-query-provider/create-table.ts

```diff
@@ -0,0 +1,71 @@
+/* eslint-disable-next-line import/no-unresolved */
+import * as AWSLambda from 'aws-lambda';
+import { Column } from '../../table';
+import { TableHandlerProps } from '../handler-props';
+import { ClusterProps, executeStatement, makePhysicalId } from './util';
+
+export async function handler(props: TableHandlerProps & ClusterProps, event: AWSLambda.CloudFormationCustomResourceEvent) {
+  const tableName = props.tableName;
+  const tableColumns = props.tableColumns;
+  const clusterProps = props;
+
+  if (event.RequestType === 'Create') {
+    await createTable(tableName, tableColumns, clusterProps);
+    return { PhysicalResourceId: makeNewPhysicalId(tableName, clusterProps, event.RequestId), Data: { tableName: tableName } };
+  } else if (event.RequestType === 'Delete') {
+    await dropTable(tableName, clusterProps);
+    return;
+  } else if (event.RequestType === 'Update') {
+    const { replace } = await updateTable(tableName, tableColumns, clusterProps, event.OldResourceProperties as TableHandlerProps & ClusterProps);
+    const physicalId = replace ? makeNewPhysicalId(tableName, clusterProps, event.RequestId) : event.PhysicalResourceId;
+    return { PhysicalResourceId: physicalId, Data: { tableName: tableName } };
+  } else {
+    /* eslint-disable-next-line dot-notation */
```

Comment:
nit: curious: why?

Reply:
Without the eslint directive, I get the following eslint error:
```
error  ["RequestType"] is better written in dot notation  dot-notation
```
If I write `event.RequestType` instead, I get the following TypeScript error:
```
error TS2339: Property 'RequestType' does not exist on type 'never'.           
```
Which would imply it recognizes that the branches above cover all possible cases. However, if I remove the `else`, then I get the following TypeScript error:
```
error TS7030: Not all code paths return a value.                                                           
```
It's a lose-lose-lose